fendermint/testing/contract-test/tests/upgrades.rs

```diff
@@ -0,0 +1,281 @@
+// Copyright 2022-2024 Protocol Labs
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+mod staking;
+
+use anyhow::Context;
+use async_trait::async_trait;
+use ethers::types::{H160, U256};
+use fendermint_contract_test::Tester;
+use fendermint_rpc::response::decode_fevm_return_data;
+use rand::rngs::StdRng;
+use rand::SeedableRng;
+use std::str::FromStr;
+
+use ethers::contract::abigen;
+use fvm_shared::address::Address;
+use fvm_shared::bigint::Zero;
+use fvm_shared::econ::TokenAmount;
+use fvm_shared::version::NetworkVersion;
+use tendermint_rpc::Client;
+
+use fendermint_crypto::SecretKey;
+use fendermint_vm_actor_interface::eam;
+use fendermint_vm_actor_interface::eam::EthAddress;
+use fendermint_vm_core::Timestamp;
+use fendermint_vm_genesis::{Account, Actor, ActorMeta, Genesis, PermissionMode, SignerAddr};
+use fendermint_vm_interpreter::fvm::store::memory::MemoryBlockstore;
+use fendermint_vm_interpreter::fvm::upgrade_scheduler::UpgradeScheduler;
+use fendermint_vm_interpreter::fvm::{
+    bundle::contracts_path, upgrade_scheduler::Upgrade, FvmMessageInterpreter,
+};
+use fendermint_vm_message::chain::ChainMessage;
+
+fn eth_addr_to_h160(eth_addr: &EthAddress) -> H160 {
+    ethers::core::types::Address::from_slice(&eth_addr.0)
+}
```

Comment:
This conversion exists as a `From` [here](https://github.com/consensus-shipyard/ipc/blob/6282ccc3f214b2de795dae4838c8f9ae7ffeea61/fendermint/vm/actor_interface/src/eam.rs#L93-L97).